### PR TITLE
Adds availableForSale to Product query fragment

### PR DESCRIPTION
**Pull request by StefanSlehta**
Addresses #594

### CODE REVIEW

**Comment by jamiemtdwyer** [comment]
LGTM, but I think this will need to be accompanied by a major version bump unless we do some mapping on the client side? I'm not sure yet how this should be handled. I'm working on replacing some other deprecated fields in parallel on another branch currently.

File: src/graphql/VariantFragment.graphql [hunk 1]
```diff
@@ -1,5 +1,6 @@
 fragment VariantFragment on ProductVariant {
   id
+  availableForSale
```
Comment (jamiemtdwyer) [inline src/graphql/VariantFragment.graphql]: we should also remove `available` here since it's deprecated, and `availableForSale` replaces it

**Comment by philspitler** [comment]
> LGTM, but I think this will need to be accompanied by a major version bump unless we do some mapping on the client side? I'm not sure yet how this should be handled. I'm working on replacing some other deprecated fields in parallel on another branch currently.

Hey @jamiemtdwyer - I don't understand how adding an extra data point to what is returned from the storefront API breaks anything.  If you are saying you are going to attach this to other breaking changes as an arbitrary dependency, I have a hard time making sense of that.  Will you please explain?

**Comment by StefanSlehta** [comment]
Hey @philspitler ! I think @jamiemtdwyer was referring to removing the deprecated `available` field as a breaking change requiring a major version bump, and not the addition of `availableForSale`

**Pull request merged**